File: TDDMicroExercises.ProposedSolution/CSharp/TurnTicketDispenser.SomeDependencies/TurnTicketAndSequenceClient.cs
using System;
namespace TDDMicroExercises.OneSolution.TurnTicketDispenser.SomeDependencies
{
    public class TurnTicketAndSequenceClient
    {
		// A class with the only goal of simulating dependencies on 
		// TurnTicket and TurnNumberSequence that have impact on the refactoring.

		public TurnTicketAndSequenceClient()
        {
			var globalSequence = TurnNumberSequence.GlobalTurnNumberSequence;
			var turnTicket1 = new TurnTicket(globalSequence.GetNextTurnNumber());
			var turnTicket2 = new TurnTicket(globalSequence.GetNextTurnNumber());
			var turnTicket3 = new TurnTicket(globalSequence.GetNextTurnNumber());
        }
    }
}

File: TDDMicroExercises/Java/TurnTicketDispenser/src/main/java/tddmicroexercises/turnticketdispenser/somedependencies/TurnTicketAndSequenceClient.java
package tddmicroexercises.turnticketdispenser.somedependencies;

import tddmicroexercises.turnticketdispenser.TurnNumberSequence;
import tddmicroexercises.turnticketdispenser.TurnTicket;

public class TurnTicketAndSequenceClient
{
    // A class with the only goal of simulating a dependencies on
    // TurnNumberSequence and TurnTicket that have impact on the refactoring.

    public TurnTicketAndSequenceClient()
    {
        TurnTicket turnTicket1 = new TurnTicket(TurnNumberSequence.getNextTurnNumber());
        TurnTicket turnTicket2 = new TurnTicket(TurnNumberSequence.getNextTurnNumber());
        TurnTicket turnTicket3 = new TurnTicket(TurnNumberSequence.getNextTurnNumber());
    }
}
